src/client/tls.rs

```diff
@@ -62,7 +49,7 @@ impl AsyncRead for MaybeTlsStream {
     }
 }
 
-impl AsyncWrite for MaybeTlsStream {
+impl<S: futures::AsyncRead + futures::AsyncWrite + Unpin + Send> futures::AsyncWrite for MaybeTlsStream<S> {
```

Comment:
You might want to use futures::{AsyncRead, AsyncWrite} to save valuable horizontal space.